Gems/ROS2/Code/Source/VehicleDynamics/DriveModels/SkidSteeringDriveModel.h

```diff
@@ -37,25 +37,43 @@ namespace ROS2::VehicleDynamics
         AZStd::pair<AZ::Vector3, AZ::Vector3> GetVelocityFromModel() override;
 
     private:
+        // Data collected once and cached for the reuse
+        struct SkidSteeringWheelData
+        {
+            WheelControllerComponent* wheelControllerComponentPtr{ nullptr };
+            WheelDynamicsData wheelData;
+            float wheelPosition{ 0.0f }; // normalized distance between the wheel and the axle's center
+            float dX{ 0.0f }; // wheel's contribution to vehicle's linear movement
+            float dPhi{ 0.0f }; // wheel's contribution to vehicle's rotational movement
+            AZ::Vector3 axis{ AZ::Vector3::CreateZero() }; // rotation axis of the wheel
+        };
+        AZStd::vector<SkidSteeringWheelData> m_wheelsCache;
+        bool m_initialized = false;
+
+        //! Collect all necessary data to compute the impact of wheels on the vehicle's velocity.
+        //! The data is stored in cache and reused.
+        void InitializeCache();
```

Comment:
Better call it ComputeWheelVelocityImpact or similar - focus on semantics (What happens when this function is called) and not implementation (Cache). Cache is a bit misleading here.